moecache.py

```diff
@@ -55,13 +56,24 @@ def pickle_dumps(obj):
     from cPickle import loads as pickle_loads
 
 except ImportError:
-    from cStringIO import StringIO as OStringStream
-    import cPickle
+    try:
+        from cStringIO import StringIO as OStringStream
+        import cPickle
 
-    def pickle_dumps(obj):
-        return cPickle.dumps(obj, 2)
+        def pickle_dumps(obj):
+            return cPickle.dumps(obj, 2)
+
+        pickle_loads = cPickle.loads
+    except ImportError:
+        from io import BytesIO as OStringStream
+        import pickle
+
+        def pickle_dumps(obj):
+            return pickle.dumps(obj, 2)
+
+        pickle_loads = pickle.loads
 
-    pickle_loads = cPickle.loads
+is_py2 = (sys.version_info[0] == 2)
```

Comment:
Make this a static private class member instead of global, and detect python 3 not 2.